src/lex.rs

```diff
@@ -237,10 +238,32 @@ pub fn lexer(input: &str, prev_ans: Option<f64>) -> Result<Vec<Token>, CalcError
             }
             'a'..='z' | 'A'..='Z' => {
                 let parse_num = num_vec.parse::<f64>().ok();
+
                 if let Some(x) = parse_num {
+                    num_vec.clear();
+
+                    // Check for exponential notation
+                    if letter == 'e' {
+                        while let Some(next) = chars.peek() {
+                            if !matches!(*next, '0'..='9') {
+                                break;
+                            }
+
+                            num_vec.push(*next);
+                            chars.next();
+                        }
+
+                        if num_vec.len() > 0 {
+                            if let Some(exp) = num_vec.parse::<usize>().ok() {
+                                result.push(Token::Num(x * pow(10 as f64, exp) as f64));
```

Comment:
Wouldn't using `num::pow` directly be better?
```suggestion
                                result.push(Token::Num(x * num::pow(10 as f64, exp) as f64));
```